### ISSUE TITLE
uvflag.to_baseline(..., force_pol=True) doesn't work

### ISSUE THREAD

**Issue by nkern** [RadioAstronomySoftwareGroup/pyuvdata#703]
I have a `UVFlag` object in `waterfall` type from IDR2.2 and I'm just trying to turn it into `baseline` type via the `to_baseline(..., force_pol=True)` because the uvf is Npols=1 and the uvd is Npols=4. Its giving the error
```
ValueError                                Traceback (most recent call last)
<ipython-input-940-2e6f6dc4abc7> in <module>
----> 1 uvf.to_baseline(uvd, force_pol=True)

~/anaconda2/envs/py36/lib/python3.6/site-packages/pyuvdata-1.4.1-py3.6-linux-x86_64.egg/pyuvdata/uvflag.py in to_baseline(self, uv, force_pol, run_check, check_extra, run_check_acceptability)
   1852             If keep_pol is False, and the original UVFlag object has more
   1853             than one polarization, the resulting polarization_array
-> 1854             will be a single element array with a comma separated string
   1855             encoding the original polarizations.
   1856         run_check : bool

~/anaconda2/envs/py36/lib/python3.6/site-packages/pyuvdata-1.4.1-py3.6-linux-x86_64.egg/pyuvdata/uvbase.py in check(self, check_extra, run_check_acceptability, ignore_requirements)
    275                                          'Parameter shape is {pshape}, expected shape is '
    276                                          '{eshape}.'.format(param=p, pshape=np.shape(param.value),
--> 277                                                             eshape=eshape))
    278                     if eshape == ():
    279                         # Single element

ValueError: UVParameter _flag_array is not expected shape. Parameter shape is (82680, 1, 1024, 4), expected shape is (82680, 1, 1024, 1).
```
which i think is a failure of the `uvf.check()` because when I check, `uvf.Npols = 1` whereas `uvf.polarization_array = np.array([-5, -6, -7, -8])`. THe issue is that the `Npols` is not appropriatley updated so when the automated type checks in `UVBase` occur the `polarization_array` fails its acceptability check. Seems like we just need to update `Npols=len(polarization_array)` in the `to_baseline()` method

**Comment by nkern** [comment]
@adampbeardsley indeed, it seems like there is no test in `test_uvflag.py` for this case. There is a test to force_pol when the polarizations do not match, but not if Npols are different.

**Comment by nkern** [comment]
addressed by #704

**Comment by bhazelton** [comment]
closed by #704
